### PR TITLE
front: tests: add tests for formatSignalingSystems function

### LINKED ISSUE #6920: front: editor: tests for formatSignalingSystems function
### Description and goal

Test formatSignalingSystems to be sure of its behavior

### Acceptance criteria

The different use cases are tested

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close #6920

### CODE REVIEW

File: front/src/applications/editor/tools/pointEdition/__tests__/utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            conditional_parameters: [],
+          },
+        ],
+      },
+      geometry: {
+        geometries: [],
+        type: 'GeometryCollection',
+      },
+    };
+    expect(formatSignalingSystems(entity)).toEqual([
```
Comment (clarani) [inline front/src/applications/editor/tools/pointEdition/__tests__/utils.spec.ts]: Modify the function and then give it only 1 logicalSignal
```suggestion
    expect(formatSignalingSystem(logicalSignal)).toEqual([
```

**Review by clarani — COMMENTED** [review]
Nice tests, thanks, it's a good start ! It would be nice to modify the function formatSignalingSystems (maybe rename it formatSignalingSystems => formSignalingSystem) so it takes only 1 logical_signal and format it as it might be easier to test.

```typescript
// test only formatSignalingSystem
export function formatSignalingSystems(
  entity: SignalEntity
): SignalEntity['properties']['logical_signals'] {
  if (!entity.properties.logical_signals) return [];
  return entity.properties.logical_signals.map((logicalSignal) => formatSignalingSystem(logicalSignal))
```

Here are the use cases I can see for now:
- handle a BAL signal without default_parameters (initialization)
- handle a BAL signal with default_parameters
- handle a BAL signal with conditional_parameters
- handle a BAL signal with default_parameters and conditional_parameters becoming a BAPR signal
- handle a BAPR signal becoming a BAL signal
- handle a modification on a BAPR signal (update of settings)

File: front/src/applications/editor/tools/pointEdition/types.ts [hunk 2]
```diff
@@ -31,6 +31,17 @@ export type SignalingSystem = {
...
+export type SignalingSystemForm = {
+  next_signaling_systems: Array<string | undefined>;
+  signaling_system: 'BAL' | 'BAPR' | 'TVM';
+  settings?: { Nf?: 'true' | 'false'; distant?: 'true' | 'false'; is_430?: 'true' | 'false' };
+  default_parameters?: { jaune_cli: 'true' | 'false' };
+  conditional_parameters: {
+    on_route?: string;
+    parameters?: { jaune_cli: 'true' | 'false' };
+  }[];
+};
```
Comment (Yohh) [inline front/src/applications/editor/tools/pointEdition/types.ts:43]: is `SignalingSystem` still relevant after adding this type?

File: front/src/applications/editor/tools/pointEdition/__tests__/utils.spec.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      signaling_system: 'BAPR',
+      settings: { Nf: 'true', distant: 'true' },
+      default_parameters: {},
+      conditional_parameters: [],
+    });
+  });
+});
+
+// TVM signaling systems were not tested, as the data system now includes TVM300 and TVM430 systems.
+// Todo - add TVM300 and TVM430 tests
```
Comment (Yohh) [inline front/src/applications/editor/tools/pointEdition/__tests__/utils.spec.ts]: some of us grep TODO to find those, it is case sensitive

```suggestion
// TODO - add TVM300 and TVM430 tests
```

**Review by Yohh — COMMENTED** [review]
nice! just a few comments

File: front/src/applications/editor/tools/pointEdition/types.ts (hunk 2 above)
Comment (Yohh) [reply front/src/applications/editor/tools/pointEdition/types.ts:43]: my bad, it is still used

File: front/src/applications/editor/tools/pointEdition/__tests__/utils.spec.ts (hunk 3 above)
Comment (Uriel-Sautron) [reply front/src/applications/editor/tools/pointEdition/__tests__/utils.spec.ts]: Done

**Review by Yohh — APPROVED** [review]
LGTM, tested :muscle:

**Review by clarani — APPROVED** [review]
LGTM, thanks :pray: 

:warning: Don't forget to squash your commits

**Pull request merged**